### PR TITLE
Fix scattergl mising points plot and react

### LINKED ISSUE #2334: Missing data in scattergl plot when I go juste above 10000 points (regression)
Hi,

I just updated plotly.js from 1.30 to the latest 1.33, and my plot is broken.

I’m rendering about 75000 points with a scattergl plot, and after the update, it looks like most of the points are not rendered at all. What is odd though, is that when I hover over the plot, I can still see popups showing when my mouse is over places where points are supposed to be.

After trying different things, I noticed everything is rendering fine up to 10000 data points. If I go just above, say 10001, my plot is broken and most of the points are not rendered!

Everything worked fine with plotly.js 1.30.

I made two versions of my code, one with 10000 points, and the other with 10001. See for yourself: https://0.x2a.yt/other/private/plotly-test/10000.html
https://0.x2a.yt/other/private/plotly-test/10001.html

(This bug report is a copy/paste from the following forum topic: https://community.plot.ly/t/regression-missing-data-in-scattergl-plot-when-i-go-juste-above-10000-points/8152)

### PR DESCRIPTION
**Pull request by archmoj**
Fix #2334, #3405 and #3751.
Bump `point cluster` 3.1.8, `regl-scatter2d` 3.1.7 and `regl-splom` 1.0.8 and update snap according to data `length`.
@plotly/plotly_js

### CODE REVIEW

File: test/jasmine/tests/scattergl_test.js [hunk 1]
```diff
@@ -583,6 +584,48 @@ describe('end-to-end scattergl tests', function() {
         .catch(failTest)
         .then(done);
     });
+
+    it('@gl should reset the sanp to length after react and not to TOO_MANY_POINTS constant', function(done) {
+        function fig(num) {
+            var x = [];
+            var y = [];
+            for(var i = 0; i < num; i++) {
+                x.push(Math.random());
```
Comment (alexcjohnson) [inline test/jasmine/tests/scattergl_test.js]: Random numbers are dangerous in tests. If you really need this to look like random numbers, you can use `Lib.pseudoRandom()` (preceded by `Lib.seedPseudoRandom()`) as a repeatable alternative.

Comment (archmoj) [reply test/jasmine/tests/scattergl_test.js]: @alexcjohnson thanks for the review.
Good call! Addressed for `scattergl_test` and other jasmine tests in https://github.com/plotly/plotly.js/pull/4323/commits/902b08491d4480fa341650dcc365b230885a0afa.

Comment (alexcjohnson) [reply test/jasmine/tests/scattergl_test.js]: Beautiful, thank you for extending that to the rest of the test suite!

File: src/traces/scattergl/calc.js [hunk 2]
```diff
@@ -97,12 +97,7 @@ module.exports = function calc(gd, trace) {
     if(opts.line && !scene.line2d) scene.line2d = true;
     if((opts.errorX || opts.errorY) && !scene.error2d) scene.error2d = true;
     if(opts.text && !scene.glText) scene.glText = true;
-
-    // FIXME: organize it in a more appropriate manner, probably in sceneOptions
-    // put point-cluster instance for optimized regl calc
-    if(opts.marker) {
-        opts.marker.snap = stash.tree || TOO_MANY_POINTS;
-    }
+    if(opts.marker) opts.marker.snap = len;
```
Comment (etpinard) [inline src/traces/scattergl/calc.js]: Ha - there is it!

That's the piece I was missing. Thank you very much @archmoj - this is probably the best bug fix for the library in 2019.

**Comment by etpinard** [comment]
I'm curious @archmoj does the new mock `gl2d_clustering` take a long time to render?

**Comment by archmoj** [comment]
> I'm curious @archmoj does the new mock `gl2d_clustering` take a long time to render?

I don't think so. Let me create a codepen to investigate the diff.
@etpinard thanks for the review.

**Pull request merged**